keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -283,6 +283,144 @@ def generate_image(
+                Default: 25.
+            unconditional_guidance_scale: float controling how closely the image
+                should adhere to the prompt. Larger values result in more
+                closely adhering to the prompt, but will make the image noisier.
+                Default: 7.5.
+            diffusion_noise: Tensor of shape (`batch_size`, img_height // 8,
+                img_width // 8, 4), or a Tensor of shape (img_height // 8,
+                img_width // 8, 4). Optional custom noise to seed the diffusion
+                process. When the batch axis is omitted, the same noise will be
+                used to seed diffusion for every generated image.
+            seed: integer which is used to seed the random generation of
+                diffusion noise, only to be specified if `diffusion_noise` is
+                None.
+        """
+        if diffusion_noise is not None and seed is not None:
+            raise ValueError(
+                "`diffusion_noise` and `seed` should not both be passed to "
+                "`generate_image`. `seed` is only used to generate diffusion "
+                "noise when it's not already user-specified."
+            )
+
+        encoded_text = self.encode_text(prompt)
+        encoded_text = tf.squeeze(encoded_text)
+        if encoded_text.shape.rank == 2:
+            encoded_text = tf.repeat(
+                tf.expand_dims(encoded_text, axis=0), batch_size, axis=0
+            )
+
+        image = tf.squeeze(image)
+        image = tf.cast(image, dtype=tf.float32) / 255. * 2. - 1.
+        image = tf.expand_dims(image, axis=0)
+        known_x0 = self.image_encoder(image)
+        if image.shape.rank == 3:
+            known_x0 = tf.repeat(known_x0, batch_size, axis=0)
+
+        mask = tf.expand_dims(mask, axis=-1)
+        mask = tf.cast(
+            tf.nn.max_pool2d(mask, ksize=8, strides=8, padding="SAME"),
+            dtype=tf.float32,
+        )
+        mask = tf.squeeze(mask)
+        if mask.shape.rank == 2:
+            mask = tf.repeat(
+                tf.expand_dims(mask, axis=0), batch_size, axis=0
+            )
+        mask = tf.expand_dims(mask, axis=-1)
+
+        context = encoded_text
+        unconditional_context = tf.repeat(
+            self._get_unconditional_context(), batch_size, axis=0
+        )
+
+        if diffusion_noise is not None:
+            diffusion_noise = tf.squeeze(diffusion_noise)
+            if diffusion_noise.shape.rank == 3:
+                diffusion_noise = tf.repeat(
+                    tf.expand_dims(diffusion_noise, axis=0), batch_size, axis=0
+                )
+            latent = diffusion_noise
+        else:
+            latent = self._get_initial_diffusion_noise(batch_size, seed)
+
+        # Iterative reverse diffusion stage
+        timesteps = tf.range(1, 1000, 1000 // num_steps)
+        alphas, alphas_prev = self._get_initial_alphas(timesteps)
+        progbar = keras.utils.Progbar(len(timesteps))
```

Comment:
Lets offer a way to turn this off to users;  my suggestions is an argument:`verbose`